Assets/Scripts/GameObjectsGenerator.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Collections;
+using UnityEngine;
+using System.Linq;
+using UnityEngine.Events;
+using Random = UnityEngine.Random;
+
+public class GameObjectsGenerator : MonoBehaviour
+{
+    [SerializeField] private float _secondsBetweenSpawn = 1f;
+    [SerializeField] private Range _range;
+    [SerializeField] private Pool _pool;
+    [SerializeField] private SpawnPointsBlackList _spawnPointBlackList;
+
+    private float _elapsedTime = 0;
+
+    public event UnityAction<Vector3> AddPositionToBlackList;
+    public event UnityAction<Vector3> CheckSpawnPosition;
```

Comment:
https://trello.com/c/CsOZfMhi/1-%D0%BD%D0%B0%D0%B7%D0%B2%D0%B0%D0%BD%D0%B8%D1%8F-%D1%81%D0%BE%D0%B1%D1%8B%D1%82%D0%B8%D0%B9-%D0%BE%D0%B1%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D1%87%D0%B8%D0%BA%D0%BE%D0%B2